src/databricks/labs/ucx/azure/access.py

```diff
@@ -25,6 +37,8 @@
 from databricks.labs.ucx.hive_metastore.locations import ExternalLocations
 
 logger = logging.getLogger(__name__)
+P = ParamSpec('P')
+R = TypeVar('R')
```

Comment:
are they even used?... 

UPDATE: found usage. define type params close to their usage, as it confuses while reading code.

Reply:
hmm, it's convention to have them as globals, so to make this makes sense. Do you want to make it part of the class then? It is above the class which uses it, with a small dataclass in between.